src/vanilla/utils/watch.ts

```diff
@@ -2,13 +2,19 @@ import { subscribe } from '../../vanilla.ts'
 
 type Cleanup = () => void
 type WatchGet = <T extends object>(proxyObject: T) => T
-type WatchCallback = (get: WatchGet) => Cleanup | void | undefined
+type WatchCallback = (
+  get: WatchGet,
+) => Cleanup | void | Promise<void> | Promise<Cleanup> | undefined
 type WatchOptions = {
   sync?: boolean
 }
 
 let currentCleanups: Set<Cleanup> | undefined
 
+function runFnCallback<T extends CallableFunction>(value: T) {
+  value()
+}
+
```

Comment:
This is just a refactor. I personally don't prefer it. Please leave them unchanged at least in this PR.
(It might be better to include `.clear`, but that should be a separate PR.)

Reply:
Fair enough. Just FYI the reasoning behind is to not create new functions on each revalidate (which can increase GC time)